### PR TITLE
Refactored the calls to list-asset, withdrawing, collateral and contributing

### PR DESCRIPTION
**Pull request by csmartinsfct**

### CODE REVIEW

File: src/apis/brain.js [hunk 1]
```diff
@@ -633,69 +633,41 @@ export const withdrawInvestorProfit = async (user, asset, notificationId, update
...
-          operationType: 'contribution',
-        },
-        status: 'info',
+export const fundAsset = async (userName, assetId, amount, onTransactionHash, onReceipt, onError) => {
+  try {
+    const fundingHubContract = new window.web3js.eth.Contract(
+      FundingHub.ABI,
+      FundingHub.ADDRESS,
+    );
+    const weiAmount = window.web3js.utils.toWei(amount.toString(), 'ether');
```
Comment (jjperezaguinaga) [inline src/apis/brain.js]: Really? Does it take Strings as a parameter?

File: src/apis/brain.js [hunk 2]
```diff
@@ -633,69 +633,41 @@ export const withdrawInvestorProfit = async (user, asset, notificationId, update
...
-          }
-          resolve(receipt.status);
-        });
-    } catch (err) {
-      resolve(false);
-    }
-  });
+const processErrorType = (error, handleError) => {
+  if(error.message.includes("User denied transaction signature")){
+    handleError('metamask');
```
Comment (jjperezaguinaga) [inline src/apis/brain.js]: Nice.

File: src/components/BlockchainInfo.js [hunk 3]
```diff
@@ -592,37 +598,77 @@ class BlockchainInfo extends React.Component {
     }
   }
 
-  async fundAsset(assetId, amount, onSuccessConfirmationPopup, onFailureContributionPopup, amountDollars) {
+  buildAndUpdateNotification(notificationId, type, status, params){
+    this.updateNotification(notificationId, {
+      [type]: {
+        ...params,
+      },
+      status,
```
Comment (jjperezaguinaga) [inline src/components/BlockchainInfo.js]: Future you will be like wat haha, jk, I think this is fine.

File: src/components/BlockchainInfo.js [hunk 4]
```diff
@@ -592,37 +598,77 @@ class BlockchainInfo extends React.Component {
...
+          onFailureContributionPopup();
+          this.buildAndUpdateNotification(notificationId, NotificationTypes.FUNDING, NotificationStatus.ERROR, {
+            assetName,
+          });
+        }
+      }
+
+      const onError = (type) => {
+        onFailureContributionPopup();
+        if(type === 'metamask'){
```
Comment (jjperezaguinaga) [inline src/components/BlockchainInfo.js]: Maybe we can have something like`ErrorTypes.METAMASK` and so on.

File: src/apis/brain.js (hunk 1 above)
Comment (csmartinsfct) [reply src/apis/brain.js]: yes, either string or BigNumber

File: src/components/BlockchainInfo.js (hunk 4 above)
Comment (csmartinsfct) [reply src/components/BlockchainInfo.js]: I knew you were gonna say something about that ahah I'll add it

File: src/components/BlockchainInfo.js (hunk 3 above)
Comment (csmartinsfct) [reply src/components/BlockchainInfo.js]: present me is already looking at the current code and having that reaction. I'm prepared for anything!

Go is getting there though

File: src/components/BlockchainInfo.js (hunk 4 above)
Comment (jjperezaguinaga) [reply src/components/BlockchainInfo.js]: You know me by now ❤️

File: src/apis/brain.js (hunk 1 above)
Comment (jjperezaguinaga) [reply src/apis/brain.js]: Jesus.

File: src/components/BlockchainInfo.js (hunk 3 above)
Comment (jjperezaguinaga) [reply src/components/BlockchainInfo.js]: 🎉

**Pull request merged**